packages/core/src/tracing/instrumentation/cloud/aws-sdk/v3/instana_aws_product.js

```diff
@@ -24,11 +24,11 @@ class InstanaAWSProduct {
    */
   getOperations() {
     const operationMap = {};
-
-    this.operations.forEach(op => {
-      operationMap[op] = this;
-    });
-
+    if (this.operations && this.operations.length) {
```

Comment:
suggestion: in line 16 do

`this.operations = operations || []`

or do

```constructor(spanName, operations: []) {```

Then you don't need this check at all.

Reply:
Thanks, updated the code with the suggestion